client/src/components/Workflow/Editor/Forms/FormTool.vue

```diff
@@ -0,0 +1,182 @@
+                    text-disable="Set at Runtime"
+                    @onChange="onChange"
+                    ref="form"
+                />
+                <FormSection
+                    :id="id"
+                    :get-node="getNode"
+                    :datatypes="datatypes"
+                    @onChange="onChangeSection"
+                    @onChangeOutputDatatype="onChangeOutputDatatype"
+                />
+            </template>
+        </FormCardTool>
+    </CurrentUser>
+</template>
+
+<script>
+import axios from "axios";
+import CurrentUser from "components/providers/CurrentUser";
+import { getAppRoot } from "onload/loadConfig";
+import Form from "components/Form/Form";
+import FormCardTool from "components/Form/FormCardTool";
+import FormSection from "./FormSection";
+import FormElement from "components/Form/FormElement";
+import { checkLabels } from "components/Workflow/Editor/modules/utilities";
+import Utils from "utils/utils";
+
+export default {
+    components: {
+        CurrentUser,
+        Form,
+        FormCardTool,
+        FormElement,
+        FormSection,
+    },
+    props: {
+        datatypes: {
+            type: Array,
+            required: true,
+        },
+        getManager: {
+            type: Function,
+            required: true,
+        },
+        getNode: {
+            type: Function,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            mainValues: {},
+            sectionValues: {},
+            messageVariant: "",
+            messageText: "",
+        };
+    },
+    computed: {
+        node() {
+            return this.getNode();
+        },
+        workflow() {
+            return this.getManager();
+        },
+        id() {
+            return `${this.node.id}:${this.node.config_form.id}`;
+        },
+        errorLabel() {
+            return checkLabels(this.node.id, this.node.label, this.workflow.nodes);
+        },
+        inputs() {
+            const inputs = this.node.config_form.inputs;
+            Utils.deepeach(inputs, (input) => {
+                if (input.type) {
+                    if (["data", "data_collection"].indexOf(input.type) != -1) {
+                        input.hiddenInWorkflow = true;
+                        input.info = `Data input '${input.name}' (${Utils.textify(input.extensions)})`;
+                        input.value = { __class__: "RuntimeValue" };
+                    } else if (!input.fixed) {
+                        input.connectable = true;
+                        input.collapsible_value = {
+                            __class__: "RuntimeValue",
+                        };
+                        input.is_workflow =
+                            (input.options && input.options.length === 0) ||
+                            ["integer", "float"].indexOf(input.type) != -1;
+                    }
+                }
+            });
+            Utils.deepeach(inputs, (input) => {
```

Comment:
Why iterate twice? Does the first iteration mutate the source structurally?

Reply:
This remained unchanged from the previous code, but I combined the two iterations into one now.